src/Hl7.Fhir.Serialization.Tests/RoundtripNewSerializers.cs

```diff
@@ -0,0 +1,220 @@
+﻿using Hl7.Fhir.Model;
+using Hl7.Fhir.Tests;
+using Hl7.Fhir.Utility;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Xml;
+
+namespace Hl7.Fhir.Serialization.Tests
+{
+    [TestClass]
+    public class RoundTripNewSerializers
+    {
+        private static readonly string jsonTestFolder = "FHIRRoundTripTestJson";
+        private static readonly string xmlTestFolder = "FHIRRoundTripTestXml";
+        private static readonly string inputFolder = "input";
+        private static readonly string intermediate1Folder = "intermediate1";
+        private static readonly string intermediate2Folder = "intermediate2";
+
+        [DynamicData(nameof(prepareExampleZipFilesXml), DynamicDataSourceType.Method, DynamicDataDisplayName = nameof(GetTestDisplayNames))]
+        [DataTestMethod]
+        [TestCategory("LongRunner")]
+        public void FullRoundtripOfAllExamplesXmlNewSerializer(string file, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            var baseTestPath = Path.Combine(Path.GetTempPath(), xmlTestFolder);
+            doRoundTrip(baseTestPath, file, xmlSerializer, xmlDeserializer, jsonOptions);
+        }
+
+        [DynamicData(nameof(prepareExampleZipFilesJson), DynamicDataSourceType.Method, DynamicDataDisplayName = nameof(GetTestDisplayNames))]
+        [DataTestMethod]
+        [TestCategory("LongRunner")]
+        public void FullRoundtripOfAllExamplesJsonNewSerializer(string file, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            var baseTestPath = Path.Combine(Path.GetTempPath(), jsonTestFolder);
```

Comment:
Can this be done in the method `prepareExampleZipFiles`? Then it is done only once, instead for every file.